react/features/authentication/middleware.ts

```diff
@@ -153,6 +158,11 @@ MiddlewareRegistry.register(store => next => action => {
             error.recoverable = true;
 
             _handleLogin(store);
+        } else {
+            batch(() => {
```

Comment:
This feeels like the wrong place to be handling this. Authentication should not be handling track stuff.

You can handle this in the media middleware, since the recoverable flag will already be set once you calll `next` when handling this action.

Reply:
But then how fo we guarantee that the media middleware will be executed after the authentication one? Aren't they executed in the order of the imports. This doesn't sound like very intuitive solution. 

Moreover the whole story is a little bit more complicated. We need to keep the gum promise only for the login dialog and in all other cases we don't need it... IMHO the whole thing doesn't look so bad to be here.